workflow/rocoto/gfs_tasks.py

```diff
@@ -86,10 +86,11 @@ def prep(self):
         gfs_enkf = True if self.app_config.do_hybvar and 'gfs' in self.app_config.eupd_cdumps else False
 
         deps = []
-        dep_dict = {'type': 'metatask', 'name': 'gdaspost', 'offset': '-06:00:00'}
+        myoffset = '-{h:02n}:00:00'.format(h=self._base["assim_freq"])
```

Comment:
I would suggest adding the following in `class Tasks` after L35 in [tasks.py](https://github.com/NOAA-EMC/global-workflow/blob/75269e4bb9764b81589efc7d703825b80e74c8f5/workflow/rocoto/tasks.py#L35)
```python3
self._base['cycling_interval'] = to_timedelta(self._base["assim_freq"])
```
and then use `[-]self._base.cycling_interval` where necessary instead of calculating `myoffset` each time.

Reply:
Thank @aerorahul for the great suggestion! I will make this change soon.

Reply:
@aerorahul I tried wxflow.timetools.to_timedelta(), it will need something as follows:
```
from wxflow import timetools
self._base['cycling_interval'] =str(timetools.to_timedelta(str(self._base['assim_freq'])+'H'))

```
This looks a little bit more complicated than using the str.format().
Hope I did not miss anything.

I will add a dictionary entry as suggested to avoid calculating `myoffset `many times but will use str.format(). Thanks!

Reply:
Current python prefers using f-strings over `.format()` or concatenation when possible:
```
from wxflow import to_timedelta
self._base['cycling_interval'] =str(to_timedelta(f'{self._base["assim_freq"]}H'))
```

wxflow also has a function to convert a timedelta to HH:MM:SS:
```
from wxflow import to_timedelta, timedelta_to_HMS
self._base['cycling_interval'] = timedelta_to_HMS(to_timedelta(f'{self._base["assim_freq"]}H'))
```

However, I don't think you need to even bother converting to a string (the default casting of a timedelta to a string is the proper format), so this probably works:
```
from wxflow import to_timedelta
self._base['cycling_interval'] = to_timedelta(f'{self._base["assim_freq"]}H')
```

Reply:
My apologies @guoqing-noaa 
I should have been more specific in my change request.
@WalterKolczynski-NOAA thank you for clarifying and providing the exact methods that perform the operations.